curvefs/src/mds/fs_manager.cpp

```diff
@@ -808,15 +808,29 @@ int FsManager::IsExactlySameOrCreateUnComplete(const std::string& fsName,
         return google::protobuf::util::MessageDifferencer::Equals(lhs, rhs);
     };
 
-    auto checkFsInfo = [fsType, volumeInfoComparator](const FsDetail& lhs,
-                                                      const FsDetail& rhs) {
+    auto s3InfoComparator = [](common::S3Info lhs, common::S3Info rhs) {
+        // for compatible with old clients
+        if (lhs.has_objectprefix() && !rhs.has_objectprefix() &&
+            lhs.objectprefix() != 0) {
+            return false;
+        }
+        if (rhs.has_objectprefix() && !lhs.has_objectprefix() &&
+            rhs.objectprefix() != 0) {
+            return false;
+        }
+
+        return google::protobuf::util::MessageDifferencer::Equals(lhs, rhs);
```

Comment:
```suggestion
        if (lhs.has_prefix() && rhs.has_prefix() && (lhs.prefix() != rhs.prefix())) {
            return false;
        }

        lhs.clear_prefix();
        rhs.clear_prefix();

        return Equals(lhs, rhs);
```

Are they equivalent? If so, this might look clearer. BTW, pls add corresponding tests.

Reply:
No, it's different. This change is mainly for compatibility with old clients which don't have object_prefix attribute. For test, I will add it later.

Reply:
> No, it's different. This change is mainly for compatibility with old clients which don't have object_prefix attribute. For test, I will add it later.

The problem is clear, and, as my understanding, we just need to ensure if both have prefix, then they must be same, otherwise, we can ignore this field.

Reply:
It can't cover the condition that the old client which doesn't have prefix, and the new client has the prefix which is 1 .